### PR TITLE
Plane: fixed relaxing of attitude controller on transition

### PR DESCRIPTION
**Pull request by tridge**
this fixes an issue found by Leonard where the attitude controller
could have residual control left over from a previous transition when
engaging the VTOL attitude controller

### CODE REVIEW

File: ArduPlane/quadplane.cpp [hunk 1]
```diff
@@ -769,6 +771,22 @@ void QuadPlane::run_z_controller(void)
...
+
+  We relax them whenever we will be using them after a period of
+  inactivity
+ */
+void QuadPlane::check_attitude_relax(void)
+{
+    uint32_t now = AP_HAL::millis();
+    if (now - last_att_control_ms > 500) {
+        attitude_control->relax_attitude_controllers();
+        attitude_control->reset_rate_controller_I_terms();
```
Comment (WickedShell) [inline ArduPlane/quadplane.cpp]: This is unneeded as it's implicitly done by `relax_attitude_controllers()`.

File: ArduPlane/quadplane.cpp [hunk 2]
```diff
@@ -1746,6 +1766,8 @@ void QuadPlane::vtol_position_controller(void)
     float ekfGndSpdLimit, ekfNavVelGainScaler;    
     ahrs.getEkfControlLimits(ekfGndSpdLimit, ekfNavVelGainScaler);
 
+    check_attitude_relax();
```
Comment (WickedShell) [inline ArduPlane/quadplane.cpp]: Should this be done before the wp nav controller gets updated in setup_target_position()? (I haven't fully chased how that interacts with pos control and attitude control)

Comment (WickedShell) [reply ArduPlane/quadplane.cpp]: This is actually inconsistently done. `vtol_position_controller` and `waypoint_controller` setup_target_position() first, while `takeoff_controller` relaxes then setups the target position.

File: ArduPlane/quadplane.cpp (hunk 1 above)
Comment (tridge) [reply ArduPlane/quadplane.cpp]: fixed

File: ArduPlane/quadplane.cpp (hunk 2 above)
Comment (tridge) [reply ArduPlane/quadplane.cpp]: I don't think it matters, but I've changed it to be consistent anyway

**Comment by WickedShell** [comment]
#9429 went in which encompassed this, it missed the removal of an unneeded controller reset, but outside of that is the same.

**Pull request closed without merging**